ddb-client/src/main/java/org/opensearch/remote/metadata/client/impl/DDBOpenSearchClient.java

```diff
@@ -218,6 +232,11 @@ public CompletionStage<PutDataObjectResponse> putDataObjectAsync(
                 item.put(RANGE_KEY, AttributeValue.builder().s(id).build());
                 item.put(SOURCE, AttributeValue.builder().m(sourceMap).build());
                 item.put(SEQ_NO_KEY, AttributeValue.builder().n(sequenceNumber.toString()).build());
+                if (request.cmkRoleArn() != null) {
```

Comment:
interesting that elsewhere you are using `!Objects.isNull()`. Is there a difference?

More interestingly, I thought earlier you implied it's only not null for OpenSearch client.  So we definitely need some clarity in the javadocs and why we're throwing an exception for non-null by default.